backend/src/services/auth/auth.service.ts

```diff
@@ -31,4 +34,102 @@ const signupLocal = async (user: UserCreateInput): Promise<AuthResponseDto> => {
   };
 };
 
-export { signupLocal };
+const requestPasswordReset = async (email: string): Promise<string> => {
+  const user = await prisma.user.findUnique({
+    where: { email },
+  });
+  if (!user) throw new Error('User with such email does not exist');
+
+  const resetToken = jwt.sign(
+    {
+      id: user.id,
+    },
+    ENV.JWT.SECRET!,
+    { expiresIn: '1h' },
+  );
+
+  await prisma.user_Security.update({
+    where: { user_id: user.id },
+    data: { password_change_token: resetToken },
+  });
+
+  const link = `localhost:${ENV.APP.SERVER_PORT}${ENV.API.V1_PREFIX}/auth/local/reset-password?token=${resetToken}`;
+
```

Comment:
We **mustn't** refer to the `localhost` from the source code. We should use the `ENV.APP.SERVER_HOST`. And the env variable in `backend.env` can be set as `localhost`